src/lib/protocols/v3/export.ts

```diff
@@ -0,0 +1,64 @@
+import { deflate } from 'pako'
+import { base64url } from 'rfc4648'
+
+import type { ProtocolVersion, ReflectionEntry } from '$lib/types'
+import { encodeInt32, encodeString, mergeTypedArrays } from '$lib/utils'
+import { PROTOCOL_VERSION } from './protocol'
+
+function encodeTime(date: Date) {
+    const timestamp = date.getTime() / 1000
+    return encodeInt32(timestamp)
+}
+
+function encodeEntry(entry: ReflectionEntry) {
+    return mergeTypedArrays(
+        encodeTime(entry.time),
+        new Uint8Array(encodeEntryData(entry.data)),
+        encodeInt32(entry.comment != null ? entry.comment.length : 0),
+        encodeString(entry.comment != null ? entry.comment : 'null'))
+}
+
+export function encodeReflectionEntries(reflections: ReflectionEntry[]) {
+    console.log('encodeReflectionEntries', reflections)
+    const encodedEntries = reflections.map(encodeEntry)
+    const data = mergeTypedArrays(encodeInt32(reflections.length), ...encodedEntries)
+    return deflate(data, { level: 9 })
+}
+
+/**
+ * Encode every pair of numbers into a one byte to compress data.
+ */
+function encodeEntryData(data: ReflectionEntry['data']) {
+    return [data[0] << 4 | data[1],
+            data[2] << 4 | data[3],
+            data[4] << 4 | data[5],
+            data[6] << 4 | data[7]]
+    
+    //const bin = data.map((number) => number.toString(2).padStart(4, '0'))
+    //return [bin[0] + bin[1], bin[2] + bin[3], bin[4] + bin[5], bin[6] + bin[7]].map((n) =>
+    //    parseInt(n, 2),
+    //)
```

Comment:
I guess these lines could just be deleted?

```suggestion
```

Reply:
Yes, I just wanted to leave them in to show the comparison with how the other versions achieve the same thing. My version avoids the need to convert to a string and then back again by using bitshifting.